src/CMakeLists.txt

```diff
@@ -20,10 +20,10 @@ add_subdirectory(model_io)
 add_subdirectory(estimation)
 add_subdirectory(solid-shapes)
 add_subdirectory(high-level)
+add_subdirectory(inverse-kinematics)
 
 if (IDYNTREE_USES_IPOPT)
   set(IDYNTREE_BUILD_IK TRUE)
-  add_subdirectory(inverse-kinematics)
 endif()
 
 if (IDYNTREE_COMPILES_OPTIMALCONTROL)
```

Comment:
IDYNTREE_USES_OPTIMALCONTROL instead ?

Reply:
I think currently the way the option are used is: `IDYNTREE_USES_{dependency_name}` (e.g. for `IDYNTREE_USES_YARP` or `IDYNTREE_USES_IPOPT`), and `IDYNTREE_COMPILES_{IDynTree component}` (e.g. [`IDYNTREE_COMPILE_TESTS`](https://github.com/robotology/idyntree/blob/master/src/CMakeLists.txt#L12)). So in this case I would say `IDYNTREE_COMPILES_OPTIMALCONTROL` is coherent with the rest of the code, even we might want to change and use always `USE`.

Anyway, this part did not change within this PR, so eventually the discussion might be moved elsewhere.